apps/projects/app/components/Shared/BountyContextMenu.js

```diff
@@ -1,14 +1,38 @@
 import React from 'react'
+import PropTypes from 'prop-types'
 import styled, { css } from 'styled-components'
 import { ContextMenuItem, GU, theme } from '@aragon/ui'
 import { usePanelManagement } from '../Panel'
 import { issueShape } from '../../utils/shapes.js'
-import { IconCoin, IconConnect, IconFile, IconView, useTheme } from '@aragon/ui'
+import { IconCoin, IconFile, IconView, useTheme } from '@aragon/ui'
 import { useAragonApi } from '../../api-react'
 
+
+const MenuItem = ({ panel, panelParams, caption, Icon }) => {
+  const theme = useTheme()
+
+  return (
+    <Item onClick={() => panel(panelParams)}>
+      <Icon color={`${theme.surfaceContent}`} />
+      <ActionLabel>
+        {caption}
+      </ActionLabel>
+    </Item>
+  )
+}
+MenuItem.propTypes = {
+  panel: PropTypes.func.isRequired,
+  caption: PropTypes.string.isRequired,
+  panelParams: PropTypes.oneOfType([
+    PropTypes.object,
+    PropTypes.arrayOf(PropTypes.object)
```

Comment:
Looks like we are only ever setting `panelParams={issue}`, so we don't need this to allow arrays. Let's not generalize our interfaces/type checking more than we need; that's a form of premature optimization. 

Also, since we only ever pass `issue` into this prop, why not just call it `issue`? We're only using `MenuItem` within this file; it doesn't need to be so general in its language.

Reply:
We've seen something similar elsewhere... But in this particular case there is one place we have:

`<MenuItem panel={allocateBounty} panelParams={[issue]} caption="Fund issue" .../>`

BountyContextMenu is used in singular issue (IssueDetails) and in Actions menu of Issues Tab, and there an array of issues is (for now) perfectly valid as parameter. Funding Panel is an exception, all other panels (apart from Curation actually) deal with single issues. So, for now this needs to stay as it is - and I called it `panelParams`, precisely because it's not always just one issue...